client/instrumenter/src/main/java/org/hawkular/apm/instrumenter/ClientManager.java

```diff
@@ -76,24 +86,40 @@ public static void initialize(Retransformer trans) {
         }
 
         if (configService != null) {
-            // Read configuration
-            CollectorConfiguration config = configService.getCollector(null, null, null, null);
-
-            if (config != null) {
-                try {
-                    updateInstrumentation(config);
-                } catch (Exception e) {
-                    System.err.println("Failed to update instrumentation rules: " + e);
-                    e.printStackTrace();
-                }
-            } else {
-                log.severe("Unable to configure APM");
-            }
+            initConfig();
         } else {
             log.info("APM has not been configured");
         }
     }
 
+    protected static void initConfig() {
+        // Read configuration
+        CollectorConfiguration config = configService.getCollector(null, null, null, null);
+
+        if (config != null) {
+            try {
+                updateInstrumentation(config);
+            } catch (Exception e) {
+                log.severe("Failed to update instrumentation rules: " + e);
```

Comment:
Can you continue here? Don't you want to bubble up the exception or even directly bail out?
If bubbling up, then don't log+throw